Add Django watchman health check to App Server

tnation14 opened the pull request:
## Overview

This PR adds a `/health-check/` endpoint to app servers so that we can detect when they're out of service. I added the endpoint and a `django-watchman` configuration that checks databases and caches.

### Checklist

- [ x ] PR has a name that won't get you publicly shamed for vagueness
~- [ ] Files changed in the PR have been `yapf`-ed for style violations~

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Once this is merged, we'll still have to turn enable Production app monitoring using Panopta.


## Testing Instructions

 * Run `docker-compose up --no-deps -d  nginx django`
 * Do `curl http://localhost:8080/health-check/ `
 * Ensure that the service returns a 500 status code
 * Run `docker-compose up -d` to start all services
 * Do `curl http://localhost:8080/health-check/ `
 * Ensure that the endpoint returns a 200 status code.

Closes #105

kshepard commented on django/publicmapping/publicmapping/urls.py:
```diff
@@ -32,6 +32,7 @@
 
 urlpatterns = ([
     url(r'^$', publicmapping_views.index),
+    url(r'^health-check/$', include('watchman.urls')),
```
I noticed a warning pass by for this line:

`Your URL pattern '^health-check/$' uses include with a regex ending with a '$'. Remove the dollar from the regex to avoid problems including URLs.`

tnation14 commented:
Updated. Warning is cleaned up and I added a Geoserver health-check as well. I added a function to  `SpatialUtils` that creates Geoserver request headers, so this PR probably needs another thorough look.

kshepard commented:
I looked through the new changes, tested it out again, and it works great. :+1:
